src/lib/block_if.c

```diff
@@ -427,23 +459,11 @@ blockif_proc(struct blockif_ctxt *bc, struct blockif_elem *be, uint8_t *buf)
 		err = block_flush(bc);
 		break;
 	case BOP_DELETE:
-#ifdef __FreeBSD__
-		if (!bc->bc_candelete)
-			err = EOPNOTSUPP;
-		else if (bc->bc_rdonly)
-			err = EROFS;
-		else if (bc->bc_ischr) {
-			arg[0] = br->br_offset;
-			arg[1] = br->br_resid;
-			if (ioctl(bc->bc_fd, DIOCGDELETE, arg))
-				err = errno;
-			else
-				br->br_resid = 0;
-		} else
-			err = EOPNOTSUPP;
-#else
-		err = EOPNOTSUPP;
-#endif
+		if (block_delete(bc, br->br_offset, br->br_resid) < 0) {
+			err = errno;
+			break;
```

Comment:
OK I've kept the function but have removed the `// off_t arg[2]`. CI is happy again (not that I expected it to object!)